src/Microsoft.Framework.PackageManager/Restore/NuGet/HttpSource.cs

```diff
@@ -131,13 +139,20 @@ internal class HttpSource
 
                 using (var response = await _client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead))
                 {
+                    if (!throwNotFound && response.StatusCode == HttpStatusCode.NotFound)
```

Comment:
For v3 feed, FindPackagesById() becomes constructing the URI by ourselves and check whether the URI points to a JSON. A 404 page means the packages is not on this feed.